services/audio_library.py

```diff
@@ -0,0 +1,175 @@
+
+        if audio_file.file.path in self.current_playbacks:
+            audio_player = self.current_playbacks[audio_file.file.path][0]
+            volume = self.current_playbacks[audio_file.file.path][1]
+
+            if fade_out_time > 0:
+                await fade_out(
+                    audio_player,
+                    volume,
+                    fade_out_time,
+                    fade_out_resolution
+                )
+            else:
+                await audio_player.stop_playback()
+
+            self.current_playbacks.pop(audio_file.file.path, None)
+
+    async def get_playback_status(self, audio_file: AudioFile|AudioFileConfig) -> [bool, AudioPlayer|None, float|None]:
+        audio_file = self.__get_audio_file_config(audio_file)
+
+        if audio_file.file.path in self.current_playbacks:
+            audio_player = self.current_playbacks[audio_file.file.path][0]
+            return [
+                audio_player.is_playing, # Is playing
+                audio_player, # AudioPlayer
+                self.current_playbacks[audio_file.file.path][1][0] # Current Volume
+            ]
+        return [False, None, None]
+
+    async def change_playback_volume(self, audio_file: AudioFile|AudioFileConfig, volume: float):
+        audio_file = self.__get_audio_file_config(audio_file)
+
+        if audio_file.file.path in self.current_playbacks:
+            self.current_playbacks[audio_file.file.path][1][0] = volume
+
+    async def on_playback_started(self, wingman_name: str):
+        self.notify_playback_started(wingman_name)
+        # Placeholder for future implementations
+        pass
+
+    async def on_playback_finish(self, wingman_name: str):
+        self.notify_playback_finished(wingman_name)
+        self.current_playbacks.pop(wingman_name, None)
+
+    def notify_playback_started(self, wingman_name: str):
+        if self.callback_playback_started:
+            # Give the callback the audio file that started playing and current volume
+            audio_file = self.current_playbacks[wingman_name][2]
+            audio_player = self.current_playbacks[wingman_name][0]
+            volume = self.current_playbacks[wingman_name][1][0]
+            self.callback_playback_started(audio_file, audio_player, volume)
+            
+    def notify_playback_finished(self, wingman_name: str):
+        if self.callback_playback_finished:
+            # Give the callback the audio file that finished playing
+            audio_file = self.current_playbacks[wingman_name][2]
+            self.callback_playback_finished(audio_file)
+
+    ### Audio Library functions ###
+
+    def get_audio_files(self) -> list[AudioFile]:
+        audio_files = []
+        try:
+            audio_files = [
+                AudioFile(path=f, name=path.basename(f))
+                for f in listdir(self.audio_library_path)
+                if path.isfile(path.join(self.audio_library_path, f))
+                and f.endswith(".wav")
```

Comment:
what about .mp3 files?

Reply:
Added them in. Did this part when I only knew that wav files are working